bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/ScriptedCustomModuleHandlerFactory.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2015-2017 Simon Merschjohann and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.module.script.rulesupport.internal;
+
+import java.util.Collection;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+
+import org.eclipse.smarthome.automation.Module;
+import org.eclipse.smarthome.automation.handler.ModuleHandler;
+import org.eclipse.smarthome.automation.handler.ModuleHandlerFactory;
+import org.eclipse.smarthome.automation.module.script.rulesupport.shared.ScriptedHandler;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ScriptedCustomModuleHandlerFactory} is used in combination with the
+ * {@link ScriptedCustomModuleTypeProvider} to allow scripts to define custom types in the RuleEngine. These
+ * registered types can then be used publicly from any Rule-Editor.
+ *
+ * This class provides the handlers from the script to the RuleEngine. As Jsr223 languages have different needs, it
+ * allows these handlers to be defined in different ways.
+ *
+ * @author Simon Merschjohann
+ *
+ */
+public class ScriptedCustomModuleHandlerFactory extends AbstractScriptedModuleHandlerFactory {
+    private Logger logger = LoggerFactory.getLogger(ScriptedCustomModuleHandlerFactory.class);
+
+    private final HashMap<String, ScriptedHandler> typesHandlers = new HashMap<>();
+
+    private ServiceRegistration<?> bmhfReg;
+
+    private static ScriptedCustomModuleHandlerFactory instance;
```

Comment:
Remove this, this is a bad smell.
Try to refactor the code so that it properly uses services instead.
It seems to be only used by `ScriptedAutomationManager` - as this is instantiated within a service, you can also inject `ScriptedCustomModuleHandlerFactory` after its instantiation.